handler/parse.go

```diff
@@ -38,24 +39,54 @@ func (d dataTree) push(k string, v []string) {
 
 // mount mounts data tree recursively.
 func (d dataTree) mount(i []string, v []string) {
-	if len(i) == 1 {
-		// single value context (last element)
-		d[i[0]] = v[len(v)-1]
+	if len(i) == 0 {
 		return
 	}
 
+	p, ok := d[i[0]]
+	if ok {
+		_, dataTreeOK := p.(dataTree)
+		if !dataTreeOK {
+			oldV := d[i[0]]
+			oldVString, stringOk := oldV.(string)
+			oldVStringArray, stringArrayOk := oldV.([]string)
+			if stringOk && len(oldVString) == 0 {
+				d[i[0]] = make(dataTree)
+			} else if stringArrayOk && len(oldVStringArray) == 0 {
+				d[i[0]] = make(dataTree)
+			} else {
+				panic(
```

Comment:
Hey @tungfinblox 👋🏻 

Thanks for the PR 👍🏻 

RR should not panic. In this case, the user may send invalid data, leading to RR shutdown.

Since we have this edge-case, it's better to return an error instead of panicking and handle it properly (return from the `request` method).

Reply:
Got it. Will send update later.

Reply:
I've just updated the PR to not panic as we discussed :+1:

Reply:
Nice, thanks 👍🏻